### PR TITLE
Streamlit App

### PR DESCRIPTION
**Pull request by aresssera**
This [Streamlit app](https://appfrictionless-test0.streamlit.app/) enables users to perform quality checks on CSV files of SFOE OGD publications using the Frictionless framework. The app allows for easy validation of CSV files against defined schemas (extracted from corresponding datapackage.json) found on uvek-gis, ensuring data integrity and adherence to specified formats.

For now, I deployed it using my personal account. The next step is to create a Streamlit Community Cloud account for one of the remaining members and deploy the app from here.

### CODE REVIEW

**Comment by AFoletti** [comment]
**Minor issue:**
The app still breaks if I try to validate a CSV using ";" instead of ",". I think it would be good to catch this error and inform the user about it instead of letting him see the app not working.

**Even more minor issue:**
You could use the "dialect" element in the datapackage.json in order to infer which separator to expect.
```
"dialect": {
                "delimiter": ",",
                "doubleQuote": true,
                "lineTerminator": "\r\n",
                "quoteChar": "\"",
                "skipInitialSpace": true,
                "header": true,
                "caseSensitiveHeader": false
            },
```

**Comment by AFoletti** [comment]
...for some reason the OGD21 CSVs return validation errors, but they look OK to me.
image

**Comment by aresssera** [comment]
> ...for some reason the OGD21 CSVs return validation errors, but they look OK to me. image


The behaviour is really strange. I made some tests with other files that contain columns of type _integer_. For example, if you take OGD105, the type of the column 'Month' is _int_, but when you access the first element of that column it says _float_.

image


I assumed that if you changed all column types to _float_, the file would be recognized as valid. To test this, I changed the type of one column and expected that this column would not be mentioned in the error message.
But surprisingly, this made the whole file valid.

image

image

**Comment by AFoletti** [comment]
Would it be possible to make a quick test with pure frictionless? I mean do not go through the pandas/numpy loop and use the full datapackage.json with pure frictionless to validate OGD21.
I clearly remember the same OGD21 pass through my early version of the tests with exactly the same datapackage. I guess that reading the file and storing it temporarily could create some sort of issue

**Comment by aresssera** [comment]
I tested `data_t1.csv` using the code from [main.py](https://github.com/SFOE/OGD_qualitychecks/blob/testing_data/main.py) (had to manually upload the file to the staging folder). It is **valid**.

File: app.py [hunk 1]
```diff
@@ -17,48 +17,48 @@ def perform_quality_check(frame, file_name):
...
                                     field['type'] = 'integer'
+
+                            # convert dictionary schema into frictionless schema object
                             schema = Schema(uploaded_file_schema)
 
                         
-
-                            
-                            print(schema)
+                            # change the data type of the first column to 'float' if all dtypes are 'int'
```
Comment (AFoletti) [inline app.py:57]: 😭

**Review by nrohrbach — COMMENTED** [review]
👍

**Pull request merged**